libs/ledger-live-common/src/families/elrond/js-getFeesForTransaction.ts

```diff
@@ -1,23 +1,14 @@
 import { BigNumber } from "bignumber.js";
-import type { ElrondAccount, Transaction } from "./types";
 import { getFees } from "./api";
-import { buildTransaction } from "./js-buildTransaction";
+import { Transaction } from "./types";
 
 /**
  * Fetch the transaction fees for a transaction
  *
- * @param {ElrondAccount} a
  * @param {Transaction} t
  */
-const getEstimatedFees = async ({
-  a,
-  t,
-}: {
-  a: ElrondAccount;
-  t: Transaction;
-}): Promise<BigNumber> => {
-  const unsigned = await buildTransaction(a, t);
-  return await getFees(JSON.parse(unsigned));
+const getEstimatedFees = async (t: Transaction): Promise<BigNumber> => {
```

Comment:
If I understood right, this will be used when function `prepareTransaction` is called, so I assume it will be used by frontends, right?
I don't see any [`AccountBridge`](https://github.com/LedgerHQ/ledger-live/blob/develop/libs/ledgerjs/packages/types-live/src/bridge.ts#L116) function related to fees, it is somehow called by frontends without the bridge?

Reply:
Not really, because this function is also called when I want to estimate the max spendable amount of the account. In this case, I do not want to build a transaction.
In addition, I would keep this logic isolated in order to extend it in the future if it will be the case.